shared/macros-oval.jinja

```diff
@@ -515,3 +515,72 @@
+#}}
+{{%- macro oval_yamlfile_value(path='', file='', yamlpath='', value='', type='', check='', negate=false, ocp_data=false) -%}}
+{{% if target_oval_version >= [5, 11] %}}
+  <def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="1">
+      <metadata>
+        <title>Check the value of '{{{ yamlpath }}}' in the "{{{ file }}}" file</title>
+        {{{- oval_affected(products) }}}
+        <description>The file {{{ file }}} in {{{ path }}} should {{% if negate %}}not{{% endif %}} contain value "{{{ value }}}" of type {{{ type }}} at {{{ yamlpath }}}.</description>
+      </metadata>
+      <criteria>
+          <criterion comment="Make sure that {{% if negate %}}no{{% endif %}} {{% if check %}}({{{ check }}}){{% endif %}} element of '{{{ yamlpath }}}' has value '{{{ value }}}'' of type {{{ type }}}." {{% if negate %}}negate="true"{{% endif %}} test_ref="test_{{{ rule_id }}}"/>
+          <criterion comment="Make sure that there is the file '{{{ file }}}' exists in {{{ path }}}." test_ref="test_file_for_{{{ rule_id }}}"/>
+      </criteria>
+    </definition>
+
+    <local_variable id="{{{ rule_id }}}_file_location" datatype="string" comment="The actual path of the file to scan." version="1">
+       {{% if ocp_data %}}
+       <concat>
+	       <variable_component var_ref="ocp_data_root"/>
+	       <literal_component>{{{ path }}}</literal_component>
+       </concat>
+       {{% else %}}
+       <literal_component>{{{ path }}}</literal_component>
+       {{% endif %}}
+    </local_variable>
+
+    <ind:yamlfilecontent_test id="test_{{{ rule_id }}}" check="at least one" comment="Find at least one match of '{{{ value }}}' in '{{{ yamlpath }}}'." version="1">
+	    <ind:object object_ref="object_{{{ rule_id }}}"/>
+	    <ind:state state_ref="state_{{{ rule_id }}}"/>
+    </ind:yamlfilecontent_test>
+
+    <unix:file_test id="test_file_for_{{{ rule_id }}}" check="only one" comment="Find the file to be checked." version="1">
+	    <unix:object object_ref="object_file_for_{{{ rule_id }}}"/>
+    </unix:file_test>
+
+    <unix:file_object id="object_file_for_{{{ rule_id }}}" version="1">
+      <unix:path var_ref="{{{ rule_id }}}_file_location"/>
+      <unix:filename>{{{ file }}}</unix:filename>
+    </unix:file_object>
+
+    <ind:yamlfilecontent_object id="object_{{{ rule_id }}}" version="1">
+      <ind:path var_ref="{{{ rule_id }}}_file_location"/>
+      <ind:filename>{{{ file }}}</ind:filename>
+      <ind:yamlpath>{{{ yamlpath }}}</ind:yamlpath>
+    </ind:yamlfilecontent_object>
+
+    <ind:yamlfilecontent_state id="state_{{{ rule_id }}}" version="1">
+	    <ind:value_of {{% if type %}}datatype="{{{ type }}}"{{% endif %}} {{% if check %}}entity_check="{{{ check }}}"{{% endif %}}>{{{ value }}}</ind:value_of>
+    </ind:yamlfilecontent_state>
+
+    <external_variable comment="Root of OCP data dump" datatype="string" id="ocp_data_root" version="1" />
```

Comment:
This probably should be wrapped in a conditional for `ocp_data`

Reply:
Right